### PR TITLE
[C-2518, C-2523, C-2611] Improve playlist create sagas

### PR DESCRIPTION
**Pull request by dylanjeffers**
### Description

Improves playlist create flow by replacing `generateRandomPlaylistId` with `getUnclaimedId`
Improves flow by moving navigation/toasts to the sagas
Improves flow by removing need to dispatch`addTrackToPlaylist` for new playlists generated by a track.
Improves saga flow by moving to new file, and naming various stages appropriately
Improves left-nav drag-drop, by disabling the existing left-nav item that is drag-n-dropping
Fixes issue where clicking left-nav always called `updateViewedAt`

### Dragons
Updated toast component to accept `link` and `linkText` props, and if they exist, replace `content` with `ToastLinkContent`. this was the only way i could get the custom toast link we needed , generated by an action. I think this is preferable, since folks are sending JSX into the actions, which are being saved in state. Added a cleanup task to remove these cases, and also clean up the implementation, but it is minor for sure.

### CODE REVIEW

File: packages/common/src/models/User.ts [hunk 1]
```diff
@@ -68,7 +68,7 @@ export type UserMetadata = {
 export type ComputedUserProperties = {
   _profile_picture_sizes: ProfilePictureSizes
   _cover_photo_sizes: CoverPhotoSizes
-  _collectionIds?: string[]
+  _collectionIds?: number[]
```
Comment (Kyle-Shanks) [inline packages/common/src/models/User.ts:71]: should this just be ID type?

Comment (dylanjeffers) [reply packages/common/src/models/User.ts:71]: def

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/dj-c-2523-improved-playlist-create

File: packages/common/src/services/audius-backend/AudiusBackend.ts [hunk 2]
```diff
@@ -3761,6 +3761,11 @@ export const audiusBackend = ({
     return audiusLibs
   }
 
+  async function getAudiusLibsTyped() {
+    await waitForLibsInit()
+    return audiusLibs as AudiusLibsType
+  }
+
```
Comment (Kyle-Shanks) [inline packages/common/src/services/audius-backend/AudiusBackend.ts:3768]: do we need a separate function for this?

File: packages/web/src/components/add-to-playlist/desktop/AddToPlaylistModal.tsx [hunk 3]
```diff
@@ -82,24 +79,16 @@ const AddToPlaylistModal = () => {
...
-        <ToastLinkContent
-          text={messages.createdToast}
-          linkText={messages.view}
-          link={playlistPage(account.handle, trackTitle, tempId)}
-        />
+      createPlaylist(
+        metadata,
+        CreatePlaylistSource.FROM_TRACK,
+        trackId,
+        'toast'
```
Comment (Kyle-Shanks) [inline packages/web/src/components/add-to-playlist/desktop/AddToPlaylistModal.tsx:89]: these are so much cleaner. I love it

File: packages/web/src/components/edit-playlist/mobile/EditPlaylistPage.tsx [hunk 4]
```diff
@@ -248,20 +243,17 @@ const EditPlaylistPage = g(
...
-        createPlaylist(tempId, formFields)
+        createPlaylist(formFields)
         toast(messages.toast)
         close()
-        goToRoute(
-          playlistPage(account.handle, formFields.playlist_name, tempId)
-        )
+        // goToRoute(
+        //   playlistPage(account.handle, formFields.playlist_name, tempId)
+        // )
```
Comment (Kyle-Shanks) [inline packages/web/src/components/edit-playlist/mobile/EditPlaylistPage.tsx:251]: we route in the saga now right? can remove this

File: packages/web/src/components/nav/desktop/PlaylistLibrary/CollectionNavItem.tsx [hunk 5]
```diff
@@ -85,6 +89,7 @@ export const CollectionNavItem = (props: CollectionNavItemProps) => {
 
   const handleDrop = useCallback(
     (draggingId: PlaylistLibraryID, kind: DragDropKind) => {
+      console.log('dropping?')
```
Comment (Kyle-Shanks) [inline packages/web/src/components/nav/desktop/PlaylistLibrary/CollectionNavItem.tsx:92]: logging?

**Review by Kyle-Shanks — APPROVED** [review]
Sweet

File: packages/common/src/services/audius-backend/AudiusBackend.ts (hunk 2 above)
Comment (dylanjeffers) [reply packages/common/src/services/audius-backend/AudiusBackend.ts:3768]: yeah sadly when i type the real one there are like 100+ type errors :/ should resolve this at some point though

**Pull request merged**